apollo-router/src/plugins/diagnostics/system_info/mod.rs

```diff
@@ -0,0 +1,631 @@
+            let display_value = if value.len() > 200 {
+                format!(
+                    "{}... (truncated, {} chars total)",
+                    &value[..200],
+                    value.len()
+                )
+            } else {
+                value
+            };
+            info.push_str(&format!("{}: {}\n", var, display_value));
+        }
+    }
+
+    // If no relevant environment variables are set, indicate that
+    if !relevant_vars.iter().any(|var| std::env::var(var).is_ok()) {
+        info.push_str("No relevant Apollo environment variables set\n");
+    }
+}
+
+/// Get normalized OS name for better cross-platform reporting
+pub(crate) fn get_normalized_os() -> &'static str {
+    match OS {
+        "apple" => "darwin",
+        "dragonfly" => "dragonflybsd",
+        "macos" => "darwin",
+        "ios" => "darwin",
+        os => os,
+    }
+}
+
+/// Get normalized architecture name for better cross-platform reporting
+pub(crate) fn get_normalized_arch() -> &'static str {
+    match ARCH {
+        "x86_64" => "amd64",
+        "aarch64" => "arm64",
+        "arm" => "arm32",
+        "powerpc" => "ppc32",
+        "powerpc64" => "ppc64",
+        arch => arch,
+    }
+}
+
+/// Detect effective CPU count considering cgroup limits (improved from fleet detector)
+#[cfg(not(target_os = "linux"))]
+fn detect_effective_cpu_count(system_cpus: u64) -> (&'static str, u64) {
+    ("system", system_cpus)
+}
+
+#[cfg(target_os = "linux")]
+fn detect_effective_cpu_count(system_cpus: u64) -> (&'static str, u64) {
+    // Determine cgroup version first
+    match std::fs::read_to_string("/proc/filesystems").map(|fs| detect_cgroup_version(&fs)) {
+        Ok(CGroupVersion::CGroup2) => {
+            // cgroup v2: read from cpu.max
+            match std::fs::read_to_string("/sys/fs/cgroup/cpu.max") {
+                Ok(contents) => {
+                    if contents.starts_with("max") {
+                        ("system", system_cpus)
+                    } else {
+                        match contents.split_once(' ') {
+                            Some((quota, period)) => (
+                                "cgroup2",
+                                calculate_cpu_count_with_default(system_cpus, quota, period),
+                            ),
+                            None => ("system", system_cpus),
+                        }
+                    }
+                }
+                Err(_) => ("system", system_cpus),
+            }
+        }
+        Ok(CGroupVersion::CGroup) => {
+            // cgroup v1: read from separate quota and period files
+            let quota = std::fs::read_to_string("/sys/fs/cgroup/cpu/cpu.cfs_quota_us")
+                .map(|s| s.trim().to_string())
+                .ok();
+            let period = std::fs::read_to_string("/sys/fs/cgroup/cpu/cpu.cfs_period_us")
+                .map(|s| s.trim().to_string())
```

Comment:
These `.to_string()`s are not needed. They are just being used as `&str`s a few lines down.